Fix to_csv of DataFrame with single level multiindex.

EricChea opened the pull request:
### What's this PR do?
Fixes the output of to_csv to show the value instead of the tuple when a ``DataFrame`` or ``Series`` is constructed using a ``MultiIndex`` with one level.  Although, I'm not sure if it's more desirable to require a MultiIndex to be instantiated with more than one level in the future. 

- [X] closes #19589
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

jreback commented on pandas/tests/io/formats/test_to_csv.py:
```diff
@@ -257,6 +257,20 @@ def test_to_csv_multi_index(self):
         exp = "foo\nbar\n1\n"
         assert df.to_csv(index=False) == exp
 
+        # GH 19589
```
make a new test

EricChea replied:
Done. 168833489d22ac79e1e214a70f93096c388c9852

jreback commented on pandas/io/formats/format.py:
```diff
@@ -1802,6 +1802,10 @@ def _save_chunk(self, start_i, end_i):
                                         date_format=self.date_format,
                                         quoting=self.quoting)
 
+        # Flatten ix when nlevels = 1 and ix is array of tuples.
```
this is not the place to fix it. it should be handled in the MI formatter itself.

EricChea replied:
I agree.  168833489d22ac79e1e214a70f93096c388c9852

jreback commented on pandas/tests/io/formats/test_to_csv.py:
```diff
@@ -257,6 +257,20 @@ def test_to_csv_multi_index(self):
+            levels=[[1.], [2.]], labels=[[0], [0]], names=['x', 'y']
+        )
+        expected1d = 'x,data\n1.0,1\n'
+        expected2d = 'x,y,data\n1.0,2.0,1\n'
```
use the pattern
```
result = 
expected = 
assert result == expected
```
use a blank line to separate cases

EricChea replied:
Thanks! Indeed it is easier to read in the suggested form. 168833489d22ac79e1e214a70f93096c388c9852

jreback commented on doc/source/whatsnew/v0.23.0.txt:
```diff
@@ -827,6 +827,7 @@ I/O
 - :class:`Interval` now supported in :func:`DataFrame.to_excel` for all Excel file types (:issue:`19242`)
 - :class:`Timedelta` now supported in :func:`DataFrame.to_excel` for xls file type (:issue:`19242`, :issue:`9155`)
 - Bug in :meth:`pandas.io.stata.StataReader.value_labels` raising an ``AttributeError`` when called on very old files. Now returns an empty dict (:issue:`19417`)
+- Bug in :func:`MultiIndex._format_native_types()`, meant to be a array of levels when level=1 instead of an array of tuples (:issue:`19589`)
```
make this what a user would want to know. don't use internal functions, this is purely a ``.to_csv()`` with a MultiIndex incorrectly printing (from a users's perspective)

EricChea replied:
Good point. a23252385e9de1a2af0832a82e168d825315d937

jreback commented on pandas/core/indexes/multi.py:
```diff
@@ -691,11 +691,15 @@ def _format_native_types(self, na_rep='nan', **kwargs):
-                        sortorder=self.sortorder, verify_integrity=False)
-
-        return mi.values
+        if len(new_levels) == 1:
```
just turn this into an Index (and call _format_native_types) on that

EricChea replied:
Gotcha.  I took this as a desire to make this clear we'll be treating this a normal index since the outputs should be the same.  a23252385e9de1a2af0832a82e168d825315d937

TomAugspurger commented:
Fixed the merge conflict, and the whatsnew note.

Ping on green.

jreback commented on doc/source/whatsnew/v0.23.0.txt:
```diff
@@ -1018,6 +1018,7 @@ I/O
 - :class:`Interval` now supported in :func:`DataFrame.to_excel` for all Excel file types (:issue:`19242`)
 - :class:`Timedelta` now supported in :func:`DataFrame.to_excel` for all Excel file types (:issue:`19242`, :issue:`9155`, :issue:`19900`)
 - Bug in :meth:`pandas.io.stata.StataReader.value_labels` raising an ``AttributeError`` when called on very old files. Now returns an empty dict (:issue:`19417`)
+- Bug in :func:`DataFrame.to_csv` where a single level MultiIndex incorrectly wrote a tuple. Now just the value of the index is written (:issue:`19589`).
```
can you move to 0.24.0

gfyoung replied:
Done.

jreback reviewed (changes requested):
can you rebase on master and will have a look

gfyoung committed:
BUG: Fix rendering of 1-level MultiIndex in to_csv

Display the level by itself instead of as a tuple
(i.e. "squash" the MultiIndex into a single level)

Closes gh-19589.

pep8speaks commented:
Hello @EricChea! Thanks for updating the PR.

 - There are no PEP8 issues in the file [`pandas/core/indexes/multi.py`](https://github.com/EricChea/pandas/blob/2dd3a1bddf36a6a093332ec948fc595b6fbf865c/pandas/core/indexes/multi.py) !

 - There are no PEP8 issues in the file [`pandas/tests/io/formats/test_to_csv.py`](https://github.com/EricChea/pandas/blob/2dd3a1bddf36a6a093332ec948fc595b6fbf865c/pandas/tests/io/formats/test_to_csv.py) !

jreback commented:
i thought the hypothesis stuff was fixed for timeout
can we do a precursor PR to address this?

gfyoung commented on pandas/core/indexes/multi.py:
```diff
@@ -732,9 +732,14 @@ def _format_native_types(self, na_rep='nan', **kwargs):
-                        sortorder=self.sortorder, verify_integrity=False)
+        if len(new_levels) == 1:
+            return Index(new_levels[0])._format_native_types()
+        else:
```
I just need to remove the hanging return statement, since we return something in both blocks.

gfyoung commented:
> i thought the hypothesis stuff was fixed for timeout

I'm pretty sure it was fixed but not sure why it's complaining:

https://github.com/pandas-dev/pandas/blob/da9e35883b28c83fcf55635f73c65bef89902f21/pandas/conftest.py#L14

cc @Zac-HD

gfyoung commented:
Hmmm...everything is green now.  Maybe Travis was just acting up.  PTAL.

Zac-HD commented:
Gah, it's *warning* about the not-yet-enforced [`deadline` setting](https://hypothesis.readthedocs.io/en/latest/settings.html#hypothesis.settings.deadline), which `-Werror` is turning into an error.  This is almost certainly just due to a slow VM on Travis, but if you want I can open a PR to fix that this evening :disappointed:

gfyoung commented:
> This is almost certainly just due to a slow VM on Travis

@Zac-HD : Indeed, that is what I suspect as well.  If you could open a PR to make our tests more robust against this, that would be great!

jreback commented:
thanks @EricChea , @Zac-HD , and @gfyoung
